project-koku/koku
# COST-1261 Process new fields in manifest

Testing: 

1. Bring everything up with Ingress 
2. Create a test customer: 
```
curl -d '{"name": "test_source", "source_type": "OCP", "authentication": {"credentials": {"cluster_id": "4e009161-4f40-42c8-877c-3e59f6baea3d"}}}' -H "Content-Type: application/json" -X POST http://0.0.0.0:8000/api/cost-management/v1/sources/ 
```
3. Upload a manifest with the old structure: 
```
curl -vvvv -F "upload=@20210726T172355-cost-mgmt.tar.gz;type=application/vnd.redhat.hccm.tar+tgz" \
                -H "x-rh-identity: eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiMDAwMDAwMSIsICJpbnRlcm5hbCI6IHsib3JnX2lkIjogIjAwMDAwMSJ9fX0=" \
                -H "x-rh-request_id: testtesttest" \
                localhost:8080/api/ingress/v1/upload
```
4. Repeat steps with a manifest with the new structure 
5. Check the db: 
```
PGPASSWORD=postgres psql postgres -U postgres -h localhost -p 15432 
postgres=# SELECT * FROM reporting_common_costusagereportmanifest;
 id |             assembly_id              |  manifest_creation_datetime   |   manifest_updated_datetime   |  manifest_completed_datetime  |  manifest_modified_datetime   | billing_period_start_datetime | num_total_files |             provider_id              | s3_csv_cleared | s3_parquet_cleared |             operator_version             | cluster_channel | operator_airgapped | operator_certified | operator_errors 
----+--------------------------------------+-------------------------------+-------------------------------+-------------------------------+-------------------------------+-------------------------------+-----------------+--------------------------------------+----------------+--------------------+------------------------------------------+-----------------+--------------------+--------------------+-----------------
  1 | ff1c03d2-e303-4ab8-a8fc-d1267bf160d4 | 2021-07-29 19:38:12.594214+00 | 2021-07-29 19:38:13.20845+00  | 2021-07-29 19:38:16.541812+00 | 2021-07-20 18:01:21.705677+00 | 2021-07-01 00:00:00+00        |               4 | 524caf52-bfb5-4373-a862-a765fb444ac0 | f              | f                  | 9530e2bfc566a513ff37aef0491101e80e626ce3 |                 |                    |                    | 
  2 | 7bf48899-adee-4dc6-9c86-44584fbece6b | 2021-07-29 19:43:10.196279+00 | 2021-07-29 19:43:10.570775+00 | 2021-07-29 19:43:13.810743+00 | 2021-07-26 17:23:55.476721+00 | 2021-07-01 00:00:00+00        |               4 | 7257b802-8ef8-45ec-8a62-25177b754635 | f              | f                  | 8c9c7b2d2729a434967a0df249cdf427e6949aae | stable-4.6      | f                  | f                  | {}
```
